clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -286,18 +424,30 @@ cache_simulator_t::check_warmed_up()
 bool
 cache_simulator_t::print_results()
 {
-    std::cerr << "Cache simulation results:\n";
-    for (unsigned int i = 0; i < knobs.num_cores; i++) {
-        print_core(i);
-        if (thread_ever_counts[i] > 0) {
-            std::cerr << "  L1I stats:" << std::endl;
-            icaches[i]->get_stats()->print_stats("    ");
-            std::cerr << "  L1D stats:" << std::endl;
-            dcaches[i]->get_stats()->print_stats("    ");
+    if (is_two_level) {
+        std::cerr << "Cache simulation results:\n";
+        for (unsigned int i = 0; i < knobs.num_cores; i++) {
+            print_core(i);
+            if (thread_ever_counts[i] > 0) {
+                std::cerr << "  L1I stats:" << std::endl;
+                l1_icaches[i]->get_stats()->print_stats("    ");
+                std::cerr << "  L1D stats:" << std::endl;
+                l1_dcaches[i]->get_stats()->print_stats("    ");
+            }
+        }
+        std::cerr << "LL stats:" << std::endl;
+        llcaches[0]->get_stats()->print_stats("    ");
```

Comment:
We could get rid of "is_two_level" and this code by adding these caches into all_caches with these names?  And having the general code below order the cache printing.